packages/forms-web-app/src/dynamic-forms/dynamic-components/list-add-more/question.js

```diff
@@ -121,19 +126,23 @@ class ListAddMoreQuestion extends Question {
 	}
 
 	/**
-	 * returns the formatted answers valuyes to be used to build task list elements
+	 * returns the formatted answers values to be used to build task list elements
 	 * @param {Object} answer
 	 * @param {Journey} journey
 	 * @param {String} sectionSegment
 	 * @returns {Array.<Object>}
 	 */
 	formatAnswerForSummary(sectionSegment, journey, answer) {
+		const isAddressQuestion = this.subQuestion instanceof AddressAddMoreQuestion;
 		let rowParams = [];
-		for (let i = 0; i < answer?.length; i++) {
+		const answerArray = isAddressQuestion
```

Comment:
this could reuse an get address function

Reply:
Added in a getAddressesForQuestion function which filters addresses based on fieldName